### PR TITLE
Add ability to update a service

### PR DESCRIPTION
**Pull request by fishnix**

### CODE REVIEW

File: orchestration/orchestration.go [hunk 1]
```diff
@@ -185,6 +173,57 @@ func (o *Orchestrator) DeleteService(ctx context.Context, input *ServiceDeleteIn
...
+							log.Errorf("failed to get task definition revisions '%s' to delete: %s", revision, err)
+							continue
+						}
+
+						deletedCredentials := make(map[string]struct{})
+						// for each task definition revision in the task definition family, delete any existing repository credentials, keeping track
+						// of ones we delete so we don't try to re-delete them.
+						// TODO: if we want to share repository credentials, we need to look for multiple container definitions using the same credentials.
+						for _, cd := range taskDefinition.ContainerDefinitions {
+							log.Debugf("cleaning '%s' container definition '%s' components", aws.StringValue(service.ServiceArn), aws.StringValue(cd.Name))
```
Comment (tenyo) [inline orchestration/orchestration.go]: 💯 Nice cleanup!

File: orchestration/orchestration.go [hunk 2]
```diff
@@ -193,16 +232,86 @@ func (o *Orchestrator) DeleteService(ctx context.Context, input *ServiceDeleteIn
...
+							newCdef.SetRepositoryCredentials(activeCdef.RepositoryCredentials)
+						}
+						break
+					}
+				}
+			}
+		}
+
+		// set cluster
+		input.ClusterName = cluster
```
Comment (tenyo) [inline orchestration/orchestration.go]: Can't we just use the existing cluster? I don't see when that would be different.

File: orchestration/orchestration.go [hunk 3]
```diff
@@ -193,16 +232,86 @@ func (o *Orchestrator) DeleteService(ctx context.Context, input *ServiceDeleteIn
...
+	if input.TaskDefinition != nil {
+		activeTdef, err := o.ECS.GetTaskDefinition(ctx, activeSvc.TaskDefinition)
+		if err != nil {
+			return nil, err
+		}
+
+		// for each container definition in the active task definition, if the active container deinition *has* repository
+		// credentials defined, check for an incoming container definition with the same name that doesn't already have the
+		// credential defined (ie. an override) and set the credentials parameter from the active container definition
+		for _, activeCdef := range activeTdef.ContainerDefinitions {
```
Comment (tenyo) [inline orchestration/orchestration.go]: Makes my job easier 👍 😄

File: orchestration/orchestration.go (hunk 2 above)
Comment (fishnix) [reply orchestration/orchestration.go]: i didn't intend for you to need to pass that attribute in the body since it's already in the URL

**Pull request merged**